axlearn/experiments/text/gpt/param_converter_test.py

```diff
@@ -0,0 +1,161 @@
+from transformers import AutoConfig
+from transformers.models.llama.modeling_llama import LlamaForCausalLM
+
+from axlearn.common import utils
+from axlearn.common.causal_lm import Model
+from axlearn.common.module import InvocationContext
+from axlearn.common.module import functional as F
+from axlearn.common.module import new_output_collection, set_current_context
+from axlearn.common.param_converter import parameters_from_llama_3
+from axlearn.common.test_utils import TestCase
+from axlearn.common.update_transformation import ForwardOutputs
+from axlearn.common.utils import NestedTensor
+from axlearn.experiments.text.gpt import c4_trainer
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+# Use cpu for the test.
+jax.config.update("jax_platform_name", "cpu")
+
+
+def compute_fuji_grad(prng_key, fuji: Model, state: NestedTensor, input_batch: NestedTensor):
+    """Compute gradient of fuji model with a pseudo loss."""
+    param_noise_key, forward_key = jax.random.split(prng_key, 2)
+
+    def _forward(model_params: NestedTensor, *, inputs: NestedTensor) -> ForwardOutputs:
+        model_params = fuji.apply_parameter_noise_recursively(
+            inputs["param_noise_key"], model_params
+        )
+        model_output_collection = new_output_collection()
+        context = InvocationContext(
+            "root",
+            parent=None,
+            module=fuji,
+            state=model_params,
+            prng_key=inputs["forward_key"],
+            output_collection=model_output_collection,
+            is_training=True,
+        )
+        with set_current_context(context):
+            _, aux = fuji(input_batch=inputs["input_batch"], return_aux=True)
+            loss = aux["logits"].mean()
+        return loss
+
+    _, grads = jax.value_and_grad(_forward, has_aux=False)(
+        state,
+        inputs=dict(
+            input_batch=input_batch,
+            forward_key=forward_key,
+            param_noise_key=param_noise_key,
+        ),
+    )
+    return grads
+
+
+def compute_llama_grad(llama, torch_ids, state):
+    """Compute the gradient of llama using a pseudo loss function.
+
+    and return the grad in the same nested dict as fuji."""
+    llama_logits = llama(torch_ids).logits
+    loss = llama_logits.mean()
+    loss.backward()
+    for p in llama.parameters():
+        p.data = p.grad.data
+    return parameters_from_llama_3(llama, state)
+
+
+class FujiConvertStateTest(TestCase):
+    @parameterized.parameters(
+        dict(
+            fuji_model_name="fuji-1B-v3",
+            llama_model_name="Llama-3.2-1B",
+        ),
+        dict(
+            fuji_model_name="fuji-3B-v3",
+            llama_model_name="Llama-3.2-3B",
+        ),
+        dict(
+            fuji_model_name="fuji-8B-v3",
+            llama_model_name="Llama-3.1-8B",
+        ),
+        dict(
+            fuji_model_name="fuji-70B-v3",
+            llama_model_name="Llama-3.1-70B",
+        ),
+    )
+    @pytest.mark.high_cpu
```

Comment:
Are high_cpu tests still run at every CI?

Reply:
I guess not. I tested them manually.